src/main/kotlin/com/makarytskyi/rentcar/service/impl/OrderServiceImpl.kt

```diff
@@ -22,69 +28,81 @@ internal class OrderServiceImpl(
     private val userRepository: UserRepository,
 ) : OrderService {
 
-    override fun getById(id: String): AggregatedOrderResponse =
-        orderRepository.findById(id)?.let { AggregatedOrderResponse.from(it) }
-            ?: throw NotFoundException("Order with id $id is not found")
-
-    override fun findAll(page: Int, size: Int): List<AggregatedOrderResponse> =
-        orderRepository.findAll(page, size).map { AggregatedOrderResponse.from(it) }
-            .toList()
-
-    override fun create(createOrderRequest: CreateOrderRequest): OrderResponse {
-        validateDates(createOrderRequest.from, createOrderRequest.to)
-        validateUserExists(createOrderRequest.userId)
-        validateCarAvailability(createOrderRequest.carId, createOrderRequest.from, createOrderRequest.to)
-
-        return OrderResponse.from(
-            orderRepository.create(CreateOrderRequest.toEntity(createOrderRequest)),
-            getCarPrice(createOrderRequest.carId)
-        )
+    override fun getById(id: String): Mono<AggregatedOrderResponse> =
+        orderRepository.findFullById(id)
+            .switchIfEmpty { Mono.error(NotFoundException("Order with id $id is not found")) }
+            .map { AggregatedOrderResponse.from(it) }
+
+    override fun findAll(page: Int, size: Int): Flux<AggregatedOrderResponse> =
+        orderRepository.findFullAll(page, size).map { AggregatedOrderResponse.from(it) }
+
+    override fun create(createOrderRequest: CreateOrderRequest): Mono<OrderResponse> {
+        return createOrderRequest.toMono()
+            .doOnNext { validateDates(it.from, it.to) }
+            .flatMap {
+                Mono.zip(
+                    Mono.defer { validateUserExists(createOrderRequest.userId) }.subscribeOn(Schedulers.parallel()),
+                    Mono.defer {
+                        validateCarAvailability(
+                            createOrderRequest.carId,
+                            createOrderRequest.from,
+                            createOrderRequest.to
+                        )
+                    }.subscribeOn(Schedulers.parallel())
+                ).thenReturn(createOrderRequest)
+            }
+            .flatMap { orderRepository.create(CreateOrderRequest.toEntity(it)) }
+            .flatMap { order -> getCarPrice(createOrderRequest.carId).map { OrderResponse.from(order, it) } }
     }
 
-    override fun deleteById(id: String) = orderRepository.deleteById(id)
+    override fun deleteById(id: String): Mono<Unit> = orderRepository.deleteById(id)
 
-    override fun findByUser(userId: String): List<OrderResponse> = orderRepository.findByUserId(userId).map {
-        OrderResponse.from(it, getCarPrice(it.carId.toString()))
-    }
+    override fun findByUser(userId: String): Flux<OrderResponse> = orderRepository.findByUserId(userId)
+        .flatMap { Mono.just(it).zipWith(getCarPrice(it.carId.toString())) }
+        .map { OrderResponse.from(it.t1, it.t2) }
```

Comment:
use destructing delcaration. It's unclear for reader what is t1 and t2

import reactor.kotlin.core.util.function.component1
import reactor.kotlin.core.util.function.component2